pkg/reader/ccip.go

```diff
@@ -313,7 +315,9 @@ func (r *ccipChainReader) ExecutedMessages(
 					Value:    0,
 					Operator: primitives.Gt,
 				}),
-				query.Confidence(primitives.Finalized),
+				// We don't need to wait for an execute state changed event to be finalized
+				// before we optimistically mark a message as executed.
+				query.Confidence(primitives.Unconfirmed),
```

Comment:
Wouldn't that break root snoozing?

Reply:
@winder are you able to chime in here? I can look into it but you might know the answer offhand :-)